From eac90d1f6e60f584fca9c762d556f676bd15d634 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:03:40 -0400
Subject: [fs] btrfs: remove useless ACL check

Message-id: <1407179109-13714-334-git-send-email-sandeen@redhat.com>
Patchwork-id: 87290
O-Subject: [RHEL7.1 PATCH 333/422] btrfs: remove useless ACL check
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: ZhangZhen <zhenzhang.zhang@huawei.com>

commit 776e4aae5508848730a82b4695edfbd7bc802c33
Author: ZhangZhen <zhenzhang.zhang@huawei.com>
Date:   Tue Apr 8 19:30:19 2014 +0800

    btrfs: remove useless ACL check

    posix_acl_xattr_set() already does the check, and it's the only
    way to feed in an ACL from userspace.
    So the check here is useless, remove it.

    Signed-off-by: zhang zhen <zhenzhang.zhang@huawei.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/acl.c b/fs/btrfs/acl.c
index 460f36b..f3e8df1 100644
--- a/fs/btrfs/acl.c
+++ b/fs/btrfs/acl.c
@@ -107,13 +107,6 @@ static int btrfs_set_acl(struct btrfs_trans_handle *trans,
  const char *name;
  char *value = NULL;
 
- if (acl) {
-  ret = posix_acl_valid(acl);
-  if (ret < 0)
-   return ret;
-  ret = 0;
- }
-
  switch (type) {
  case ACL_TYPE_ACCESS:
   name = POSIX_ACL_XATTR_ACCESS;
-- 
1.7.1